### PR TITLE
os/bluestore: clear up redundant size assignment in KerenelDevice

### PR DESCRIPTION
**Pull request by shashalu**
Signed-off-by: Shasha Lu <lu.shasha@eisoo.com>

### CODE REVIEW

**Comment by smithfarm** [comment]
Jenkins re-test this please

File: src/os/bluestore/KernelDevice.cc [hunk 1]
```diff
@@ -130,7 +130,6 @@ int KernelDevice::open(const string& p)
   } else {
     size = st.st_size;
   }
-  size &= ~(block_size);
```
Comment (xiexingguo) [inline src/os/bluestore/KernelDevice.cc]: The block_size can be reconfigured and hence can be any value other than st.st_blksize(see L116).
So we still need to align to block_size here. right?

Comment (shashalu) [inline src/os/bluestore/KernelDevice.cc]: yes, it should align to block_size.  It should be `size &= ~(block_size - 1)` .
L157 do this correctly, so this line is redundant.

Comment (xiexingguo) [inline src/os/bluestore/KernelDevice.cc]: > It should be size &= ~(block_size - 1)

Quite right. Looks good to me!

**Comment by liewegas** [comment]
http://pulpito.ceph.com/sage-2017-07-07_20:50:14-rados-wip-sage-testing-distro-basic-smithi/

**Pull request merged**